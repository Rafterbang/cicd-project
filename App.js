import React from "react";
import { View, Text } from "react-native";

const sum = (a, b) => {
  return a + b;
};

const App = () => {
  const result = sum(3, 5);

  return (
    <View>
      <Text>Result: {result}</Text>
    </View>
  );
};

export default App;
