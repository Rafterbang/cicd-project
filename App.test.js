import React from "react";
import { create } from "react-test-renderer";
import App from "./App";

describe("App component", () => {
  test("sum function adds 3 + 5 to equal 8", () => {
    const component = create(<App />);
    const instance = component.root;

    // Find the Text component and extract its children (the result)
    const resultText = instance.findByType("Text").props.children;

    // Extract the number from the resultText (e.g., "Result: 8" -> 8)
    const result = parseInt(resultText.split(" ")[1]);

    expect(result).toBe(8);
  });
});
