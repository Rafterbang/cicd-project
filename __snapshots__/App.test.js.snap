// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "marginBottom": 20,
      }
    }
  >
    Count: 
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="incrementButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="decrementButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement
      </Text>
    </View>
  </View>
</View>
`;
